app/home.component.ts

```diff
@@ -9,21 +9,19 @@ export class HomeComponent {
 
   vm: HomeViewModel = null;
 
-  view: null;
-
   constructor() {
-    topic.subscribe('map-view', this.onMapView.bind(this));
+    this.vm = new HomeViewModel({});
   }
 
-
-  onMapView(view) {
-    this.vm = new HomeViewModel({
-      view: view
-    });
+  setView(view) {
+    this.vm.set('view', view);
```

Comment:
yes, we can. I've tested it, and it works. All staged and ready to push. any other changes before I do?